app/code/Magento/Inventory/Controller/Adminhtml/Source/SourceRegionDataProcessor.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Controller\Adminhtml\Source;
+
+/**
+ * Prepare region data. Specified for form structure
+ */
+class SourceRegionDataProcessor
+{
+    /**
+     * @param array $data
+     * @return array
+     */
+    public function process(array $data): array
+    {
+        if (isset($data['region_id']) && '' === $data['region_id']) {
```

Comment:
There is case when we can get both `region` and `region_id` and in this case `region`  should be destroyed.